src/Marten/Linq/Parsing/SelectTransformBuilder.cs

```diff
@@ -104,6 +104,12 @@ public string ToJsonBuildObjectPair(IFieldMapping mapping, ISerializer serialize
                         locator = field.TypedLocator;
                     }
 
+                    if (field.FieldType.IsClass && field.FieldType != typeof(string) && field.FieldType != typeof(decimal))
```

Comment:
@egorpavlikhin any specific reason why you are doing these additional checks i.e. `&& field.FieldType != typeof(string) && field.FieldType != typeof(decimal)`?

Reply:
And I'm gonna weigh in too. I'd much rather that this test be baked into the `IField` model rather than being a runtime test here. The Linq provider gets very unruly and hard to support when we've used copious amounts of "if/then" logic on exceptions like this.

Reply:
This has hung out too long. We'll do Linq code cleanup later whenever we inevitably tackle more enhancements to the Select() functionality.